src/algorand_blocks/state_proof_tracking.rs

```diff
@@ -0,0 +1,82 @@
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    algorand_errors::AlgorandError,
+    algorand_types::{Result, Bytes},
+};
+
+fn is_zero(num: &Option<u64>) -> bool {
```

Comment:
These functions are being copied everywhere. Where could we put them?
Also the new one for checking the empty hashmap...

Reply:
Probably best in a `./src/utils.rs` or similar mod. Just make sure they're only public within the crate since we don't want them in the external API.

Reply:
Done, nice.